tests/integrators/maxwell_curl.cc

```diff
@@ -0,0 +1,115 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2012 - 2015 by the deal.II authors
+//
+// Author: Jihuan Tian <jihuan_tian@hotmail.com>
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+// Test curl related functions in integrators/maxwell.h
+
+#include <deal.II/base/logstream.h>
+#include <cstring>
+#include <fstream>
+#include <iostream>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/tria_iterator.h>
+#include <deal.II/grid/tria_accessor.h>
+
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_accessor.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/fe_values.h>
+
+#include <deal.II/lac/full_matrix.h>
+
+#include <deal.II/integrators/maxwell.h>
+
+using namespace dealii;
+using namespace LocalIntegrators::Maxwell;
+
+template <int dim>
+void make_grid(Triangulation<dim> &tr)
+{
+  GridGenerator::subdivided_hyper_cube(tr, 1, -1., 1.);
+
+  std::ofstream out ("grid.eps");
+  GridOut grid_out;
+  grid_out.write_eps (tr, out);
+  std::cout << "Grid written to grid.eps" << std::endl;
+}
+
+template <int dim>
+void TestMaxwellCurl(Triangulation<dim> &tr)
+{
+  int element_order = 1;
+  int quadrature_order = 3;
+
+  DoFHandler<dim> dof_handler(tr);
+  FESystem<dim> fe(FE_Q<dim>(element_order), dim);
+  FEValues<dim> fe_values(fe, QGauss<dim>(quadrature_order), update_values | update_JxW_values | update_gradients | update_quadrature_points);
+  FEFaceValues<dim> fe_face_values(fe, QGauss<dim-1>(quadrature_order), update_values | update_JxW_values | update_gradients | update_normal_vectors | update_quadrature_points);
+
+  dof_handler.distribute_dofs(fe);
+
+  const unsigned int dofs_per_cell = fe.dofs_per_cell;
+
+  FullMatrix<double> curl_curl_check(dofs_per_cell, dofs_per_cell);
+  FullMatrix<double> curl_check(dofs_per_cell, dofs_per_cell);
+  FullMatrix<double> nitsche_curl_check(dofs_per_cell, dofs_per_cell);
+
+  curl_curl_check = 0;
+  curl_check = 0;
+  nitsche_curl_check = 0;
+  
```

Comment:
The automatic indentation check complains that you have a non-empty line 80. Could you please fix that? (I think that is the only issue. In the future, you can avoid problems by running `astyle`, version 2.04, on the code you want to commit, see here https://dealii.org/developer/doxygen/deal.II/CodingConventions.html - the simple way is to run the script `contrib/utilities/indent` that runs astyle on all files.)